flytekit/core/type_engine.py

```diff
@@ -969,22 +969,43 @@ def get_literal_type(self, t: Type[T]) -> Optional[LiteralType]:
         except Exception as e:
             raise ValueError(f"Type of Generic List type is not supported, {e}")
 
+    @staticmethod
+    def isBatchable(t: Type, FlytePickle: Type[T]):
+        print("hi!!!!!!!!!!!!!!",)
+        if hasattr(t, "__origin__") and t.__origin__ == FlytePickle:
+            return True
+        elif get_origin(t) is not None:
+            return any(map(lambda x: ListTransformer.isBatchable(x, FlytePickle), get_args(t)))
+        else:
+            return False
+
     def to_literal(self, ctx: FlyteContext, python_val: T, python_type: Type[T], expected: LiteralType) -> Literal:
+        from flytekit.types.pickle import FlytePickle
         if type(python_val) != list:
             raise TypeTransformerFailedError("Expected a list")
 
         t = self.get_sub_type(python_type)
-        lit_list = [TypeEngine.to_literal(ctx, x, t, expected.collection_type) for x in python_val]  # type: ignore
+        if self.isBatchable(python_type, FlytePickle):
```

Comment:
should use snake case
```suggestion
        if self.is_batchable(python_type, FlytePickle):
```